app/controllers/cases_controller.rb

```diff
@@ -152,45 +152,43 @@ def new
     end
   end
 
-  def create #rubocop:disable Metrics/MethodLength
-    set_correspondence_type(params.fetch(:correspondence_type))
-    case_params = create_params(@correspondence_type_key)
-
-    case_class_service = GetCaseClassFromParamsService.new(
-      type: @correspondence_type,
-      params: params["case_#{@correspondence_type_key}"]
-    )
-    case_class_service.call()
-    if case_class_service.error?
-      permitted_correspondence_types
-      prepare_new_case
-      @case.assign_attributes(case_params)
-      case_class_service.set_error_on_case(@case)
-      render(:new)
-    else
-      case_class = case_class_service.case_class
-      authorize case_class, :can_add_case?
-
-      service = CaseCreateService.new current_user, case_class, case_params
+  def create
+    begin
+      set_correspondence_type(params.fetch(:correspondence_type))
+      service = CaseCreateService.new current_user, @correspondence_type_key, params
+      authorize service.case_class, :can_add_case?
       service.call
       @case = service.case
       case service.result
       when :assign_responder
         flash[:creating_case] = true
-        flash[:notice] = "#{@case.type_abbreviation} case created<br/>Case number: #{@case.number}".html_safe
+        flash[:notice] = service.flash_notice
         redirect_to new_case_assignment_path @case
       else # including :error
         @case = @case.decorate
         @case_types = @correspondence_type.sub_classes.map(&:to_s)
         @s3_direct_post = s3_uploader_for @case, 'requests'
         render :new
       end
+    rescue ActiveRecord::RecordNotUnique
+      flash[:notice] = t('activerecord.errors.models.case.attributes.number.duplication')
+      render :new
     end
+  end
 
-  rescue ActiveRecord::RecordNotUnique
-    flash[:notice] =
-      t('activerecord.errors.models.case.attributes.number.duplication')
-    render :new
+  def new_overturned_ico
```

Comment:
I see that the functionality required here is quite unique to Overturned ICO cases ... but I'm just a bit concerned about having case-specific methods at the controller level like this. I feel like anything case-specific should be in a service or a concern. Can we try to move this out at some point, not necessarily for this PR?

Reply:
I did think about this.  The problem is I think that this is different from any other new action - it takes a parameter(the id of the ICO appeal from which it is created).  I'll have a play around and see if that causes any problems.